source/_posts/2024-06-26-voice-chapter-7.markdown

```diff
@@ -0,0 +1,123 @@
+
+ - Turn off the lights in 5 minutes
+ - Pause TV in 10 minutes
+ - Open the blinds in 5 minutes
+
+Large language models (LLMs) are also able to create delayed commands, for instance saying "I'm going to bed early tonight", might lead the LLM to send itself a delayed command to start the bedtime routine sooner.
+
+## AI in control
+
+Speaking of LLMs, our Home Assistant 2024.6 update allows [AI agents to control your devices](/blog/2024/06/05/release-20246/#voice--assist). This is part of our broader AI strategy we outlined [in a recent blog](/blog/2024/06/07/ai-agents-for-the-smart-home/). It will be an important part of the future smart home, and Home Assistant will be the best platform for AI because of its focus on privacy and choice. We know it is not for everyone, nor is it ready for mass adoption, but we are making it available to experiment with. Give it a try on your voice hardware today, with device control currently available on both [Google AI](/integrations/google_generative_ai_conversation/) and [OpenAI](/integrations/openai_conversation/).
+
+## Expand an LLM with scripts
+
+To make it easier to expand the things that LLMs can do, Home Assistant 2024.7 (coming next week), will allow LLMs to access approved scripts. This lets you precisely control what the LLM will do in specific situations while retaining flexibility and natural language processing AI's are so good at.
+
+<img src='/images/blog/2024-06-voice-chapter-7/voice-script.png' alt="Exposing voice scripts to Assist in Home Assistant Voice settings" class='no-shadow'>
+
+For example, telling an LLM conversation agent that you're leaving the house may result in some unexpected behavior, such as turning off devices you want to be left on. To fix this, create a "leaving home" script that does exactly what you want and expose it to Assist. You could also add fields into the script, which help further guide the LLM, like giving it different behaviors for short or long trips away from home. Now when you say something like "I'm heading out for a couple of minutes", using the script it will lock the door, but not turn off the air conditioning.
+
+## Enhanced Media Control
+
+Assist can now control media players with simple commands like "pause", "resume","next", and "set volume to 100%". While simple on the surface, a lot of complexity is hidden underneath. Consider a living room with two media players: a smart speaker that is paused and a TV that is playing. If the user says "pause", the TV will be paused since it is the only media player that is currently playing. However, if the user then says "resume", Assist will only resume the TV because it was the last media player paused.
```

Comment:
```suggestion
Assist can now control media players with simple commands like "pause", "resume", "next", and "set volume to 100%". While simple on the surface, a lot of complexity is hidden underneath. Consider a living room with two media players: a smart speaker that is paused and a TV that is playing. If the user says "pause", the TV will be paused since it is the only media player that is currently playing. However, if the user then says "resume", Assist will only resume the TV because it was the last media player paused.
```